rust/src/lib.rs

```diff
@@ -3237,6 +3238,156 @@ pub unsafe extern "C" fn zcashlc_set_transaction_status(
+    /// transparent address is requested.
+    ///
+    /// Fully transparent transactions, and transactions that do not contain either shielded inputs
+    /// or shielded outputs belonging to the wallet, may not be discovered by the process of chain
+    /// scanning; as a consequence, the wallet must actively query to find transactions that spend
+    /// such funds. Ideally we'd be able to query by [`OutPoint`] but this is not currently
+    /// functionality that is supported by the light wallet server.
+    ///
+    /// The caller evaluating this request on behalf of the wallet backend should respond to this
+    /// request by detecting transactions involving the specified address within the provided block
+    /// range; if using `lightwalletd` for access to chain data, this may be performed using the
+    /// [`GetTaddressTxids`] RPC method. It should then call [`wallet::decrypt_and_store_transaction`]
+    /// for each transaction so detected.
+    ///
+    /// [`GetTaddressTxids`]: crate::proto::service::compact_tx_streamer_client::CompactTxStreamerClient::get_taddress_txids
+    SpendsFromAddress {
+        address: *mut c_char,
+        block_range_start: u32,
+        /// An optional end height; no end height is represented as `-1`
+        block_range_end: i64,
+    },
+}
+
+/// A struct that contains a pointer to, and length information for, a heap-allocated
+/// slice of [`FfiTransactionDataRequest`] values.
+///
+/// # Safety
+///
+/// - `ptr` must be non-null and must be valid for reads for `len * mem::size_of::<FfiTransactionDataRequest>()`
+///   many bytes, and it must be properly aligned. This means in particular:
+///   - The entire memory range pointed to by `ptr` must be contained within a single allocated
+///     object. Slices can never span across multiple allocated objects.
+///   - `ptr` must be non-null and aligned even for zero-length slices.
+///   - `ptr` must point to `len` consecutive properly initialized values of type
+///     [`FfiTransactionDataRequest`].
+/// - The total size `len * mem::size_of::<FfiTransactionDataRequest>()` of the slice pointed to
+///   by `ptr` must be no larger than isize::MAX. See the safety documentation of pointer::offset.
+/// - See the safety documentation of [`FfiTransactionDataRequest`]
+#[repr(C)]
+pub struct FfiTransactionDataRequests {
+    ptr: *mut FfiTransactionDataRequest,
+    len: usize, // number of elems
+}
+
+impl FfiTransactionDataRequests {
+    pub fn ptr_from_vec(v: Vec<FfiTransactionDataRequest>) -> *mut Self {
+        let (ptr, len) = ptr_from_vec(v);
+        Box::into_raw(Box::new(FfiTransactionDataRequests { ptr, len }))
+    }
+}
+
+/// Frees an array of FfiTransactionDataRequest values as allocated by `zcashlc_transaction_data_requests`.
+///
+/// # Safety
+///
+/// - `ptr` must be non-null and must point to a struct having the layout of [`FfiTransactionDataRequests`].
```

Comment:
Technically we allow `ptr` to be null, but if it is non-null it must...

Reply:
```suggestion
/// - `ptr` if `ptr` is non-null it must point to a struct having the layout of [`FfiTransactionDataRequests`].
```